be/src/storage/persistent_index.cpp

```diff
@@ -1931,8 +1931,9 @@ Status ShardByLengthMutableIndex::flush_to_immutable_index(const std::string& pa
             const auto nbucket = MutableIndex::estimate_nbucket(key_size, size, nshard, npage_hint);
             const auto expand_exponent = nshard / shard_size;
             for (auto i = 0; i < shard_size; ++i) {
+                // if keep_delete == true, flush immutable index with Delete Flag
                 RETURN_IF_ERROR(_shards[shard_offset + i]->flush_to_immutable_index(writer, expand_exponent, npage_hint,
```

Comment:
Why to pass not keep_delete?

Reply:
```
    virtual Status flush_to_immutable_index(std::unique_ptr<ImmutableIndexWriter>& writer, size_t nshard,
                                            size_t npage_hint, size_t nbucket, bool without_null) const = 0;
```
Because this parameter is `without_null`, so `keep_delete == true` means `without_null == false`.

Reply:
It's better to use a positive nono. Like `enable` not `disable`, `with` not `without`.

Reply:
Done